parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,336 @@
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTest.CoreSqlParsingError.UNPARSED_DOES_NOT_MATCH_ORIGINAL_ERROR;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTest.CoreSqlParsingError.UNPARSING_ERROR;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTest.DatabaseDialect.ALL;
+import static java.lang.Math.min;
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static java.nio.file.FileVisitResult.TERMINATE;
+import static java.nio.file.Files.createFile;
+import static java.nio.file.Files.isHidden;
+import static java.nio.file.Files.readAllLines;
+import static java.nio.file.Files.walkFileTree;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public final class SqlLogicTest
+{
+    private static final Logger LOG = Logger.get(SqlLogicTest.class);
+    private final String nameOfResultFile;
+    private final SqlLogicTestResult result;
+    private final FileVisitor<Path> fileVisitor;
+    private final Path testFilesRootPath;
+    private final Path resultPath;
+    private final DatabaseDialect databaseDialect;
+    private final int maxNumFilesToVisit;
+
+    enum CoreSqlParsingError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ORIGINAL_ERROR
+    }
+
+    enum DatabaseDialect
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
+
+    @Inject
+    public SqlLogicTest(SqlLogicTestConfig config)
+    {
+        this.nameOfResultFile = requireNonNull(config.getNameOfResultFile(), "name of result file is null");
+        this.testFilesRootPath = requireNonNull(config.getTestFilesRootPath(), "test files root path is null");
+        this.resultPath = requireNonNull(config.getResultPath(), "result path is null");
+        this.databaseDialect = requireNonNull(config.getDatabaseDialect(), "database dialect is null");
+        this.result = new SqlLogicTestResult(config.getDatabaseDialect());
+        this.fileVisitor = new SqlLogicTestFileVisitor();
+        this.maxNumFilesToVisit = config.getMaxNumFilesToVisit();
+    }
+
+    public static void execute()
+            throws IOException
+    {
+        Bootstrap app = new Bootstrap(new SqlLogicTestModule());
+
+        Injector injector = app
+                .doNotInitializeLogging()
+                .quiet()
+                .initialize();
+
+        injector.getInstance(SqlLogicTest.class).run();
+    }
+
+    private void run()
+            throws IOException
+    {
+        createFileForResult(resultPath, nameOfResultFile);
+        walkFileTree(testFilesRootPath, fileVisitor);
+        writeTestResultToFile(result, resultPath, nameOfResultFile);
+    }
+
+    private static void createFileForResult(Path resultPath, String nameOfResultFile)
+            throws IOException
+    {
+        try {
+            createFile(resultPath.resolve(nameOfResultFile));
```

Comment:
Do we need both resultPath and nameOfResultFile? If we always resolve nameOfResultFile against resultPath then maybe just have one config for it.

Reply:
Agreed. A config parameter for nameOfResultFile seems unnecessary, so I've made it a static final variable in SqlLogicTest. 

Done.